### PR TITLE
Improve component docs

### PR DESCRIPTION
**Pull request by Scotchester**
## Additions

- A new how-to on creating a new component
- Instructions on using wagtail-inventory in the data migration how-to

## Changes

- General tidying-up on the `atomic-structure.md` and `editing-components.md`

## Testing

1. Pull branch
1. Follow the steps at https://cfpb.github.io/cfgov-refresh/contributing-docs/#running-the-docs-locally
1. View the two changes pages

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

File: docs/editing-components.md [hunk 1]
```diff
@@ -32,9 +32,10 @@ _* If you're going to be doing anything more than making minor updates to
         1. [Adding CSS](#adding-css)
         1. [Adding JavaScript](#adding-javascript)
 1. [How-to guides](#how-to-guides)
-    1. [Adding a field](#adding-a-field)
-    1. [Editing a field](#editing a field)
-    1. [Removing a field](#removing-a-field)
+    1. [Creating a new component](#creating-a-new-component)
+    1. [Adding a field](#adding-a-field-to-an-existing-component)
```
Comment (chosak) [inline docs/editing-components.md]: Minor suggestion: can these TOC entries contain the full title of the section, e.g. "Adding a field to an existing component?"

File: docs/editing-components.md [hunk 2]
```diff
@@ -335,7 +336,26 @@ that includes the `RelatedContent` molecule in one of its StreamFields.
 ## How-to guides
 
 
-### Adding a field
+### Creating a new component
+
+1. Review the [Notes on Atomic Design](../atomic-structure/) page.
+1. Add each of the parts mentioned above:
+   1. [Create the Python class](#the-python-class)
```
Comment (chosak) [inline docs/editing-components.md]: If this is intended to be a sub-list, [it needs to be indented by four spaces](https://commonmark.org/help/tutorial/10-nestedLists.html). Currently this shows up as one list for me.

File: docs/editing-components.md [hunk 3]
```diff
@@ -421,6 +441,40 @@ In other words, if an existing field or block is changing,
...
+1. With a current database dump, visit http://localhost:8000/admin/inventory/.
+   (This is also available from the admin menu by going to
+   **Settings** > **Block Inventory**.)
+1. Using the three pairs of dropdown menus at the top of the page,
+   choose to look for pages that either
+   _include_ or _exclude_ particular components.
+    - If multiple components are selected, resulting pages must
+      match _all_ of the conditions.
+1. Click **Find matching pages** to execute the search.
+1. If no results are found, lucky you!
```
Comment (chosak) [inline docs/editing-components.md]: What about preserving legacy uses of blocks on non-production systems, e.g. build?

File: docs/editing-components.md (hunk 1 above)
Comment (Scotchester) [reply docs/editing-components.md]: Sure thing.

File: docs/editing-components.md (hunk 2 above)
Comment (Scotchester) [reply docs/editing-components.md]: Gah. That MkDocs markdown renderer strikes again!

File: docs/editing-components.md (hunk 3 above)
Comment (Scotchester) [reply docs/editing-components.md]: I do not want to encourage anyone to think of stuff on Build as permanent.

**Comment by Scotchester** [comment]
@chosak Issues addressed, rebased and force-pushed.

**Pull request merged**